src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs

```diff
@@ -0,0 +1,128 @@
+﻿//  
+// Copyright (c) Microsoft.  All rights reserved.
+// 
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands
+{
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
+    using System;
+    using System.Management.Automation;
+
+    [Cmdlet(VerbsCommon.Set, Constants.ApiManagementBackend)]
+    [OutputType(typeof(PsApiManagementBackend))]
+    public class SetAzureApiManagementBackend : AzureApiManagementCmdletBase
+    {
+        [Parameter(
```

Comment:
With this many named parameters, the strong suggestion is to add a new parameter set that that accepts a PSApiManagementBackEnd as one of the parameters, and use ValueFromPipeline there instead.  This is not *required*, but it is a suggestion.